dot-net/Centroid/Config.cs

```diff
@@ -78,7 +78,23 @@ public override bool TryConvert(ConvertBinder binder, out object result)
 
             try
             {
-                result = Convert.ChangeType(RawConfig, binder.Type);
+                JToken rawConfig = RawConfig;
+                result = rawConfig.ToObject(binder.Type);
+                //ensure at least one property in the config bound to the type
```

Comment:
I'd like to see some tests that cover this bit of logic. I played with it without looking at this code and derived this rule, but it seems like a non-obvious constraint.